nifi-nar-bundles/nifi-jmx-bundle/nifi-jmx-processors/src/main/java/org/apache/nifi/processors/jmx/GetJMX.java

```diff
@@ -0,0 +1,385 @@
+            .description("Include only these MBean domain(s) and type(s).  Domains are comma delimited "
+                    + "and optional MBean types follow a colon and are space delimited.   "
+                    + "Format: [domain1[:type1[ type2]][,domain2]]")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .required(false)
+            .build();
+
+    public static final PropertyDescriptor BLACKLIST = new PropertyDescriptor
+            .Builder().name("DomainBlackList")
+            .description("Include everything excluding these MBean domain(s) and type(s).   Domains are "
+                    + "comma delimited and optional MBean types follow a colon and are space delimited.   "
+                    + "Format: [domain1[:type1[ type2]][,domain2]])")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .required(false)
+            .build();
+
+    public static final PropertyDescriptor POLLING_INTERVAL = new PropertyDescriptor.Builder()
+            .name("Polling Interval")
+            .description("Indicates how long to wait before performing a directory listing")
+            .required(true)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .defaultValue("300 sec")
+            .build();
+
+    public static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+            .name("Batch Size")
+            .description("The maximum number of files to pull in each iteration")
+            .required(true)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("100")
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("All FlowFiles that are created are routed to this relationship")
+            .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("FlowFiles that encounter errors are routed to this relationship")
+            .build();
+
+    private List<PropertyDescriptor> descriptors;
+
+    private Set<Relationship> relationships;
+    private final BlockingQueue<Record> queue = new LinkedBlockingQueue<>();
+
+    private final Set<Record> inProcess = new HashSet<>();    // guarded by queueLock
+    private final Set<Record> recentlyProcessed = new HashSet<>();    // guarded by queueLock
+
+    private final Lock queueLock = new ReentrantLock();
+    private final Lock listingLock = new ReentrantLock();
+    private final AtomicLong queueLastUpdated = new AtomicLong(0L);
+
+    private final JsonBuilderFactory jsonBuilderFactory = Json.createBuilderFactory(null);
+
+    private ListFilter whiteListFilter;
+    private ListFilter blackListFilter;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
```

Comment:
@olegz once gave me a hint that there's an issue with it since init() method as well as few others are invoked more then once per life-cycle of a processor (see https://issues.apache.org/jira/browse/NIFI-1318).

Consider using a static initializer